### PR TITLE
Change sample row click behavior

### PR DESCRIPTION
**Pull request by gregdingle**
# Description

image

The current behavior is seriously annoying me and others because missing a checkbox opens the sample page and resets all checkboxes. The screenshot above shows the new target area. The rest of the row now selects the checkbox, which is easy to undo if unintentional. 

# Notes

* I did not try to fix the current bug where click drag selecting text causes `handleRowClick` to fire
* The meaning of our row click stats will change

# Tests

* Click on new target area, see sample page open
* Click on row, see checkbox

### CODE REVIEW

File: app/assets/src/components/views/discovery/DiscoveryView.jsx [hunk 1]
```diff
@@ -1036,7 +1036,6 @@ class DiscoveryView extends React.Component {
                 onMapLevelChange={this.handleMapLevelChange}
                 onMapMarkerClick={this.handleMapMarkerClick}
                 onMapTooltipTitleClick={this.handleMapTooltipTitleClick}
-                onSampleSelected={this.handleSampleSelected}
```
Comment (gregdingle) [inline app/assets/src/components/views/discovery/DiscoveryView.jsx]: `handleSampleSelected` is still used for search

File: app/assets/src/components/views/discovery/DiscoveryView.jsx [hunk 2]
```diff
@@ -1210,18 +1209,19 @@ class DiscoveryView extends React.Component {
...
-                  dim => dim.values,
-                  keyBy("dimension", dimensions)
-                )}
-                {...filters}
-                domain={domain}
-                onFilterChange={this.handleFilterChange}
-                allowedFeatures={allowedFeatures}
-              />
-            )}
+            {showFilters &&
```
Comment (gregdingle) [inline app/assets/src/components/views/discovery/DiscoveryView.jsx]: this is just an incidental js prettier change

**Comment by gregdingle** [comment]
That's a good alternative @tfrcarvalho . Gmail also has hover effects (which IDseq lacks) to make it obvious what the effect of the click will be. 

image

**Comment by tfrcarvalho** [comment]
Additionally, we can consider persisting the checkbox selection (although this can be tricky) and if we go with click on row to select i would get rid of the box of the checkbox (just use the shaded checkmark when hovering).

**Comment by happyimadesignr** [comment]
Hey @gregdingle, glad you're working on this! We should just expand to allow the invisible area around the checkbox to be a part of the checkbox click. It's more intuitive for the rest of the row to navigate you to the sample report.

**Comment by gregdingle** [comment]
> allow the invisible area around the checkbox to be a part of the checkbox click

Thanks @happyimadesignr . Could you provide some guidance on how much area around the checkbox and whether there is an existing element that covers it?

**Comment by happyimadesignr** [comment]
@gregdingle sure ill provide a mock

**Comment by happyimadesignr** [comment]
Screen Shot 2020-01-15 at 2 32 31 PM

Here you go! That area around the checkbox, it'll help as they're clicking down that column. I like your suggestion to have a hover state, that will help, this hover state is aligned with the design system work!

**Comment by jshoe** [comment]
is this one still in progress?

**Comment by gregdingle** [comment]
I'm going to close this PR because I don't want to make it a priority right now. I only posted the PR because I hit the issue myself and I saw an easy fix. Anyone is welcome to pick up the JIRA story: https://jira.czi.team/browse/IDSEQ-1700 .

**Pull request closed without merging**